File: CPP/ParkingLot.cpp
#pragma once
//pemanggilan library yang ingin dipakai
#include <iostream>
#include <string>
#include <list>
#include "Car.cpp"
#include "Motorcycle.cpp"



using namespace std;
//pengisian class 
class ParkingLot{
    private:
    int kapasitas;
    int jumlah_kendaraan_saat_ini;
    list<Car> mobil_parkir;     //komposit dari kelas mobil dalam bentuk list
    list<Motorcycle> motor_parkir;  //komposit dari kelas motor dalam bentuk list

    public:
    ParkingLot(){
    }

    //konstruktor ber parameter
    ParkingLot(int kapasitas, int jumlah_kendaraan_saat_ini, list<Car> mobil_parkir, list<Motorcycle> motor_parkir) {
        this->kapasitas = kapasitas;
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
        this->mobil_parkir = mobil_parkir;
        this->motor_parkir = motor_parkir;
    }
    //getter setter method
    void setKapasitas(int kapasitas){
        this->kapasitas = kapasitas;
    }

    int getKapasitas(){
        return kapasitas;
    }

    void setJumlah_kendaraan_saat_ini(int jumlah_kendaraan_saat_ini){
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
    }

    int getJumlah_kendaraan_saat_ini(){
        return jumlah_kendaraan_saat_ini;
    }

    void setMobil_parkir(list<Car> mobil_parkir){
        this->mobil_parkir = mobil_parkir;
    }

    list<Car> getMobil_parkir(){
        return mobil_parkir;
    }

    void setMotor_parkir(list<Motorcycle> motor_parkir){
        this->motor_parkir = motor_parkir;
    }

    list<Motorcycle> getMotor_parkir(){
        return motor_parkir;
    }

    ~ParkingLot(){
    }
};

File: PYTHON/ParkingLot.py
#pengisian kelas parking lot
class ParkingLot:
    #konstruktor berparameter
    def __init__ (self, kapasitas, jumlah_kendaraan_saat_ini):
        self.kapasitas = kapasitas
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini
        self.mobil_parkir = [] #contoh penggunaan composite yang berbentuk list
        self.motor_parkir = [] #contoh penggunaan composite yang berbentuk list
        
    #getter dan setter
    def setKapasitas(self, kapasitas):
        self.kapasitas = kapasitas
        
    def getKapasitas(self):
        return self.kapasitas
    
    def setJumlah_kendaraan_saat_ini(self, jumlah_kendaraan_saat_ini):
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini
        
    def getJumlah_kendaraan_saat_ini(self):
        return self.jumlah_kendaraan_saat_ini
    
    def setMobil_parkir(self, mobil_parkir):
        self.mobil_parkir = mobil_parkir
        
    def getMobil_parkir(self):
        return self.mobil_parkir
    
    def setMotor_parkir(self, motor_parkir):
        self.motor_parkir = motor_parkir
        
    def getMotor_parkir(self):
        return self.motor_parkir